src/Element.cc

```diff
@@ -222,24 +228,19 @@ void Element::Copy(const ElementPtr _elem)
   for (Param_V::iterator iter = _elem->dataPtr->attributes.begin();
        iter != _elem->dataPtr->attributes.end(); ++iter)
   {
-    if (!this->HasAttribute((*iter)->GetKey()))
+    if (this->HasAttribute((*iter)->GetKey()))
     {
-      this->dataPtr->attributes.push_back((*iter)->Clone());
+      this->RemoveAttribute((*iter)->GetKey());
     }
-    ParamPtr param = this->GetAttribute((*iter)->GetKey());
-    (*param) = (**iter);
+    auto clonedAttribute = (*iter)->Clone();
+    clonedAttribute->SetParentElement(shared_from_this());
+    this->dataPtr->attributes.push_back(clonedAttribute);
   }
 
   if (_elem->GetValue())
   {
-    if (!this->dataPtr->value)
-    {
-      this->dataPtr->value = _elem->GetValue()->Clone();
-    }
-    else
-    {
-      *(this->dataPtr->value) = *(_elem->GetValue());
-    }
+    this->dataPtr->value = _elem->GetValue()->Clone();
```

Comment:
`Clone` always allocates memory, so it's more expensive than assignment. Is there a reason for changing the logic here?

Reply:
Thanks for catching this, reverted here, https://github.com/ignitionrobotics/sdformat/pull/589/commits/03bf815f5610d2f3ac44cf54b7a51f042b9d5b38